simplestore/src/main/java/com/uber/simplestore/impl/SimpleStoreImpl.java

```diff
@@ -191,6 +194,63 @@
         orderedIoExecutor);
   }
 
+  @Override
+  public ListenableFuture<Object> deleteAllNow() {
+    SimpleStoreFactory.flushAndClearRecursive(this);
+
+    return Futures.submitAsync(
+        () -> {
+          if (scopedDirectory != null) {
+            recursiveDelete(scopedDirectory);
+          }
+          return Futures.immediateFuture(null);
+        },
+        orderedIoExecutor);
+  }
+
+  /** Only call from the orderedIoExecutor. */
+  void clearCache() {
+    cache.clear();
+  }
+
+  /**
+   * Cause all items in the queue to fail out, then run something before enabling the queue again
+   */
+  void flushQueueThenRun(Exception exception, Runnable runnable) {
```

Comment:
"flush" to me implies that the remaining jobs in will be forced to run. Maybe "cancelPendingThenRun"?